config/plugin_selector_test.go

```diff
@@ -75,6 +90,133 @@ func TestSetPluginLabelSelections(t *testing.T) {
 	}
 }
 
+func TestKeyRegex(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   string
+		wantErr bool
+	}{
+		// Valid
+		{name: "simple", input: "abc"},
+		{name: "alphanumeric", input: "abc123"},
+		{name: "with-dash", input: "abc-123"},
+		{name: "with-dot", input: "abc.123"},
+		{name: "with-underscore", input: "abc_123"},
+		{name: "mixed", input: "A_z-9.X"},
+		{name: "single-char", input: "a"},
+		{name: "long-key", input: "abc.def-ghi_jkl.mno_123"},
+		{name: "starts-with-dot", input: ".abc"},
+		{name: "ends-with-dot", input: "abc."},
+		{name: "two-dots", input: "a..b"},
+		// Invalid
+		{name: "empty", input: "", wantErr: true},
+		{name: "wildcard-star", input: "abc*", wantErr: true},
+		{name: "wildcard-question", input: "abc?", wantErr: true},
+		{name: "space", input: "abc def", wantErr: true},
+		{name: "unicode", input: "ümlaut", wantErr: true},
+		{name: "symbols", input: "abc$", wantErr: true},
+		{name: "slash", input: "abc/def", wantErr: true},
+		{name: "colon", input: "abc:def", wantErr: true},
+		{name: "comma", input: "abc,def", wantErr: true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := CheckSelectionKeyValuePairs(tt.input, "value")
+			if tt.wantErr {
+				require.Error(t, err)
+			}
+		})
+	}
+}
+
+func TestSelectorValueRegex(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   string
+		wantErr bool
+	}{
+		// Valid
+		{name: "simple", input: "abc"},
+		{name: "with-wildcards", input: "a*b?c"},
+		{name: "only-star", input: "*"},
+		{name: "only-question", input: "?"},
+		{name: "mixed-wildcards", input: "*a?b*c*"},
+		{name: "alphanumeric", input: "abc123"},
+		{name: "combo", input: "A_z-9.X*?foo"},
+		{name: "dash-dot-underscore-wildcards", input: "a_b-c.d*?"},
+		{name: "ends-with-wildcard", input: "abc*"},
+		{name: "starts-with-wildcard", input: "*abc"},
+		{name: "wildcard-middle", input: "ab*cd?ef"},
+		{name: "long-value", input: "abc.def-ghi*jkl?mno_123"},
+
+		// Invalid
+		{name: "empty", input: "", wantErr: true},
+		{name: "space", input: "abc def", wantErr: true},
+		{name: "unicode", input: "ümlaut", wantErr: true},
+		{name: "control-char", input: "abc\x00def", wantErr: true},
+		{name: "symbols", input: "abc$", wantErr: true},
+		{name: "slash", input: "abc/def", wantErr: true},
+		{name: "colon", input: "abc:def", wantErr: true},
+		{name: "comma", input: "abc,def", wantErr: true},
+		{name: "plus", input: "abc+def", wantErr: true},
+		{name: "pipe", input: "abc|def", wantErr: true},
+		{name: "caret", input: "abc^def", wantErr: true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := CheckSelectionKeyValuePairs("key", tt.input)
+			if tt.wantErr {
+				require.Error(t, err)
+			}
```

Comment:
Also need to check false positives..
```suggestion
			if tt.wantErr {
				require.Error(t, err)
			} else {
				require.NoError(t, err)
			}